rest-api/vip-endpoints.php

```diff
@@ -123,6 +131,94 @@ public function list_sites() {
 
 		return new WP_REST_Response( $sites );
 	}
+
+	/**
+	 * Build list of active plugins on site ()
+	 */
+	public function list_plugins() {
+		$all_plugins = array();
+		$standard_plugins = array();
+		$shared_plugins = array();
+		$mu_plugins = array();
+		$client_mu_plugins = array();
+
+		if ( ! function_exists( 'get_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		// array of all standard plugins
+		$standard_plugins = get_plugins();
+		$tmp_plugins = array();
+		foreach ( $standard_plugins as $key => $plugin ) {
+			if ( is_plugin_active( $key ) ) {
+				$tmp_plugins[ $key ] = array(
+					'name' => $plugin['Name'],
+					'version' => $plugin['Version'],
+					'description' => $plugin['Description'],
+					'type' => 'standard',
+				);
+			}
+		}
+		$all_plugins['standard'] = $tmp_plugins;
+
+		// array of all mu plugins
+		$mu_plugins = get_mu_plugins();
+		$tmp_plugins = array();
+		foreach ( $mu_plugins as $key => $plugin ) {
+			$tmp_plugins[ $key ] = array(
+				'name' => $plugin['Name'],
+				'version' => $plugin['Version'],
+				'description' => $plugin['Description'],
+				'type' => 'mu-plugin',
+			);
+		}
+		$all_plugins['mu-plugin'] = $tmp_plugins;
+
+		// array of all client mu plugins
+		$client_mu_plugins = get_client_mu_plugins();
+		$tmp_plugins = array();
+		foreach ( $client_mu_plugins as $key => $plugin ) {
+			$tmp_plugins[ $key ] = array(
+				'name' => $plugin['Name'],
+				'version' => $plugin['Version'],
+				'description' => $plugin['Description'],
+				'type' => 'client-mu-plugin',
+			);
+		}
+		$all_plugins['client-mu-plugin'] = $tmp_plugins;
+
+		// array of all shared plugins (activated via code and via UI)
+		if ( class_exists( 'WPCOM_VIP_Plugins_UI' ) ) {
+			$tmp_ui_plugins = array();
+			$tmp_code_plugins = array();
+			$vip_plugins = WPCOM_VIP_Plugins_UI::instance();
+			$shared_plugins = $vip_plugins->get_shared_plugins();
+
+			foreach ( $vip_plugins->get_shared_plugins() as $key => $plugin ) {
+				if ( $t = $vip_plugins->is_plugin_active( basename( dirname( $key ) ) ) ) {
+					if ( 'manual' === $t ) {
+						$tmp_code_plugins[ $key ] = array(
+							'name' => $plugin['Name'],
+							'version' => $plugin['Version'],
+							'description' => $plugin['Description'],
+							'type' => 'vip-shared-code',
+						);
+					} else {
+						$tmp_ui_plugins[ $key ] = array(
+							'name' => $plugin['Name'],
+							'version' => $plugin['Version'],
+							'description' => $plugin['Description'],
+							'type' => 'vip-shared-ui',
+						);
+					}
+				}
+			}
+			$all_plugins['vip-shared-code'] = $tmp_code_plugins;
+			$all_plugins['vip-shared-ui'] = $tmp_ui_plugins;
+		}
+
+		return new WP_REST_Response( array( 'site' => get_home_url(), 'plugins' => $all_plugins ) );
```

Comment:
Why do we need to include the URL in the response? Making a call to the API would require knowing it already.